docs/software_requirements/interrupts.sdoc

```diff
@@ -1,62 +1,204 @@
 [DOCUMENT]
 TITLE: Interrupts
-REQ_PREFIX: ZEP-
+REQ_PREFIX: ZEP-SRS-7-
 
 [GRAMMAR]
 IMPORT_FROM_FILE: software_requirements.sgra
 
 [REQUIREMENT]
-UID: ZEP-58
+UID: ZEP-SRS-7-1
 STATUS: Draft
 TYPE: Functional
 COMPONENT: Interrupts
-TITLE: Service routine for handling interrupts (ISR)
+TITLE: Installing static IRQ service routines (ISR).
 STATEMENT: >>>
-The Zephyr RTOS shall provide support a service routine for handling interrupts (ISR).
+Zephyr RTOS shall provide a mechanism to initialize a static IRQ service routine (ISR),
+providing all parameters needed to configure the hardware and software.
 <<<
+RELATIONS:
+- TYPE: Parent
+  VALUE: ZEP-39
 
 [REQUIREMENT]
-UID: ZEP-59
+UID: ZEP-SRS-7-2
 STATUS: Draft
 TYPE: Functional
 COMPONENT: Interrupts
-TITLE: Multi-level interrupts
+TITLE: Static IRQ initial status.
 STATEMENT: >>>
-The Zephyr RTOS shall support multi-level preemptive interrupt priorities, when supported by hardware. Note: detailed analysis to demonstrate non interference will be needed here.
+The static IRQ shall be initially disabled.
 <<<
+RELATIONS:
+- TYPE: Parent
+  VALUE: ZEP-39
 
 [REQUIREMENT]
-UID: ZEP-60
+UID: ZEP-SRS-7-3 
 STATUS: Draft
 TYPE: Functional
 COMPONENT: Interrupts
-TITLE: Associating application code with interrupts
+TITLE: Installing low-power and low-latency direct IRQ service routines (ISR).
 STATEMENT: >>>
-The Zephyr RTOS shall provide an interface for associating application code with specific interrupts. (CLARIFY: Can it be a deferred procedure call at interrupt context? Would be different requirement)
+Zephyr RTOS shall provide a mechanism to initialize a direct IRQ handler,
+providing all parameters needed to configure the hardware and software.
```

Comment:
What about the low power and low latency which was mentioned in the Title?

Reply:
A Direct IRQ Handler allows customizations to the IRQ handler via macros that allow developer defined abbreviated setup and teardown code specific to the use-case being supported.  Also allows developer to enable, or not, exiting of CPU LowPowerMode.  See ZEP-39-1, Direct ISR, Platform Specific helpers.

The "standard" IRQ installer implementation includes general purpose setup and teardown code, which is necessarily much slower. And always exits LPM. 

All this is designed to enable low power and fast interrupts without modifying the HAL.  But still requires someone at in the organization to write the setup, teardown, and lpm code fragments that are correct for the platform.  The alternative is to modify the HAL.  There are trade-offs.

Reply:
Will be cleaner if Title is changed to:
Installing Direct IRQ serivce routines (ISR).

Reply:
> Will be cleaner and more correct if Title is changed to: Installing Direct IRQ service routines (ISR).

That is a good point I think that change clarify this topic good